tests/phpunit/CRM/Core/DAOTest.php

```diff
@@ -598,4 +598,41 @@ public function testUpdateTimestampWithBlankDate() {
     $this->assertNull($dao->expired_date);
   }
 
+  public function testHtmlEscapeInput() {
+    $contact = new CRM_Contact_DAO_Contact();
+    $contact->contact_type = 'Individual';
+    $contact->first_name = 'Greater > Than';
+    $contact->last_name = 'Less < Than';
+    $contact->middle_name = '&';
+    $contact->save();
```

Comment:
Fun fact I learned while writing this test - I expected this test to pass but it doesn't, because `$dao->save()` does not actually escape-on-input as I previously thought. That actually happens in the form layer; `postProcess()` functions get values back from the form escaped, and then pass those through to `$dao->save()`.

Reply:
Yeah, no, that should fail... DAO has had other bits of evil (like `'null'` and "don't save timestamps"), but the escape-on-input evil was a higher-level thing (QuickForm and API). DAO saving and fetching are just, well, pleasantly ordinary here...

Does that changed your POV on whether DAO is a good level to do filtering?

Reply:
> Does that changed your POV on whether DAO is a good level to do filtering?

No, I'm sold on this solution. I think it gives a very good facsimile of the legacy behavior to keep our old code humming along while freeing our new code from the weirdness. And the test fail is only temporary; once the packages PR is merged, this too shall pass.